app/experimenter/experiments/tests/api/v5/test_mutation.py

```diff
@@ -0,0 +1,50 @@
+import json
+
+from django.conf import settings
+from django.urls import reverse
+from graphene_django.utils.testing import GraphQLTestCase
+
+
+class TestCreateExperiment(GraphQLTestCase):
+    GRAPHQL_URL = reverse("nimbus-api-graphql")
+
+    def test_create_experiment(self):
+        user_email = "user@example.com"
+        response = self.query(
+            """
+            mutation($input: CreateExperimentMutationInput!) {
+                createExperiment(input: $input) {
+                    nimbusExperiment {
+                        status
+                        name
+                        slug
+                    }
+                    errors {
+                        field
+                        messages
+                    }
+                    clientMutationId
+                }
+            }
+            """,
+            variables={
+                "input": {
+                    "name": "test1234",
+                    "slug": "slug1234",
+                    "clientMutationId": "randomid",
+                }
+            },
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+        result = content["data"]["createExperiment"]
+        experiment = result["nimbusExperiment"]
+        self.assertEqual(
+            experiment, {"status": "DRAFT", "name": "test1234", "slug": "slug1234"}
+        )
+
+        self.assertEqual(result["clientMutationId"], "randomid")
+
+        errors = result["errors"]
+        self.assertEqual(errors, [])
```

Comment:
We would also expect that the model is created here right?  If so then we should also look up the model instance and check that the fields are set.

Also in general will the GQL interface use slugs as primary identifiers for everything since we hid the db ids?  How will that work for creating related models, like branches?

Reply:
Returning the model requires it to save correctly, so I figured if the result came back it must be in the database or there would've been an error. I can add a bit to test that it went into the db as expected, at that point we are testing that library code from graphene-django works as advertised though.

Yes, its assumed we're using slugs as primary identifiers. I'm not sure what you mean about how it works for branches.

Reply:
We can worry about the branch stuff later.  And yeah I agree it's testing the internals but in general for this kind of behaviour, ie hit the API with some data, expect a change to the db, I would always still explicitly assert the db changes to test the integration.  Ie if the form were changed from a ModelForm to a Form, it could still pass this test but not do any db inserts.

Reply:
Okey, added the db lookup to verify it was added.